src/macaron/repo_finder/provenance_extractor.py

```diff
@@ -232,3 +233,133 @@ def _clean_spdx(uri: str) -> str:
     """
     url, _, _ = uri.lstrip("git+").rpartition("@")
     return url
+
+
+def _check_if_input_repo_commit_provenance_conflict(
```

Comment:
Why is this function marked to be private? Since it is called from the `analyzer` module, I think it shouldn't be private.